### PR TITLE
Add support to PyProxy for using async python generators from js

### PR DESCRIPTION
**Pull request by hoodmane**
This is on top of #3289. It adds support for using async Python generators via PyProxy.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

File: src/core/pyproxy.ts [hunk 1]
```diff
@@ -938,6 +1026,43 @@ export class PyProxyIteratorMethods {
   }
 }
 
+export class PyProxyAsyncIteratorMethods {
+  async next(arg: any = undefined): Promise<IteratorResult<any, any>> {
```
Comment (ryanking13) [inline src/core/pyproxy.ts:1030]: It seems like `PyProxy.next` is shown twice in docs (async and sync version), but I'm not sure what we should do. So I'm okay to merge.

**Review by ryanking13 — APPROVED** [review]
Thanks @hoodmane!

File: src/core/pyproxy.ts (hunk 1 above)
Comment (hoodmane) [reply src/core/pyproxy.ts:1030]: I think we can add `/** @private */` to one.

Comment (hoodmane) [reply src/core/pyproxy.ts:1030]: In the long run it'd probably be good to break up the `PyProxy` docs like the way I did to the `JsProxy` docs in #3277.

Comment (ryanking13) [reply src/core/pyproxy.ts:1031]: > I think we can add `/** @private */` to one.

That makes sense. Thanks!

Comment (hoodmane) [reply src/core/pyproxy.ts:1031]: Of course the docs aren't strictly accurate since `next` might instead translate to Python `anext`. But it is okay.

**Comment by hoodmane** [comment]
Thanks for the review @ryanking13!

**Pull request merged**